### PR TITLE
Added extra function to return function return value from chat

### PR DESCRIPTION
**Pull request by samuelgjekic**
Im not sure if this was needed, but just in case i added **generateChatOrReturnFunction** based on your **generateTextOrReturnFunction** method, i needed it anyways in my project.  Feel free to delete this pull request if this was not needed. 

      ###  Modified files:
      1. 	modified:   src/Chat/ChatInterface.php
      2. 	modified:   src/Chat/OpenAIChat.php

### CODE REVIEW

File: src/Chat/OpenAIChat.php [hunk 1]
```diff
@@ -99,6 +99,23 @@ public function generateChat(array $messages): string
...
+    {
+        $openAiArgs = $this->getOpenAiArgs($messages);
+        $answer = $this->client->chat()->create($openAiArgs);
+        $toolsToCall = $this->getToolsToCall($answer);
+
+        foreach ($toolsToCall as $toolToCall) {
+            $this->lastFunctionCalled = $toolToCall;
+        }
+
+        if ($this->lastFunctionCalled instanceof FunctionInfo) {
```
Comment (MaximeThoonsen) [inline src/Chat/OpenAIChat.php:112]: Hey @samuelgjekic,
I think it is great to add this possibility.

what if the previous answer was a function and now it is a text but the lastFunctionCalled is still the FunctionInfo from the previous call?

Comment (samuelgjekic) [reply src/Chat/OpenAIChat.php:114]: Hmm i did not think about this, my bad.. Anyways, does this not apply to the generateTextOrReturnFunction aswell? As they are the same. 

I added a line that sets the lastFunction to null in the start of the method, is this good, or do you prefer that i check if **$toolstocall** is not null instead? If so dont merge yet and i will do it that way instead.

**Comment by samuelgjekic** [comment]
@MaximeThoonsen I added the new GPT4Omni but i forgot i had this pull request so they ended up in the same, i added a solution ( i think? ) to the lastFunction above.

**Comment by MaximeThoonsen** [comment]
hey @samuelgjekic, thanks a lot for the PR! Can you please rebase your branch with main so I can review only the file that changed?

**Pull request closed without merging**